include/alpaka/block/shared/dyn/BlockSharedMemDynBoostAlignedAlloc.hpp

```diff
@@ -89,9 +93,16 @@ namespace alpaka
                             block::shared::dyn::BlockSharedMemDynBoostAlignedAlloc const & blockSharedMemDyn)
                         -> T *
                         {
+                            static_assert(
+                                core::vectorization::defaultAlignment >= alignof(T),
```

Comment:
Why do we have this requirement?

Reply:
The block shared memory is allocated with this alignment (see above). Accessing wrongly aligned types is not possible. We could increase this alignment so that it is good for all types.